authorize_helper.go

```diff
@@ -84,10 +84,10 @@ func MatchRedirectURIWithClientRedirectURIs(rawurl string, client Client) (*url.
 			// If no redirect_uri was given and the client has exactly one valid redirect_uri registered, use that instead
 			return redirectURIFromClient, nil
 		}
-	} else if rawurl != "" && isMatchingRedirectURI(rawurl, client.GetRedirectURIs()) {
+	} else if redirectTo, ok := isMatchingRedirectURI(rawurl, client.GetRedirectURIs()); rawurl != "" && ok {
```

Comment:
I think `rawurl != ""` should be first.

Reply:
It's unfortunately not possible to check before define so unless this is rewritten without `else if` it's the only possibility :/